frontend/src/components/editor/Cell.tsx

```diff
@@ -409,7 +409,10 @@ const CellComponent = (
         />
       </div>
       <OutputArea
+        // Only allow expanding in edit mode
         allowExpand={editing}
+        // Force expand when markdown is hidden
+        forceExpand={isMarkdownCodeHidden}
```

Comment:
this only forces it to be expand when its `markdown + hidden`, instead of just `markdown`. we could easily change this to just markdown if that turns out to be more preferred